### PR TITLE
Update intake form benefit types for updated VA Form 20-0995 and 20-0996

### PR DESCRIPTION
**Pull request by meckila**
Resolves [CASEFLOW-2260](https://vajira.max.gov/browse/CASEFLOW-2260) and [CASEFLOW-2261](https://vajira.max.gov/browse/CASEFLOW-2261) 

### Description

- Update Supplemental Claim intake form to use '"Veterans Readiness and Employment" instead of "Vocational Rehabilitation and Employment"
- Update Higher Level Review intake form to use "Veterans Readiness and Employment" instead of "Vocational Rehabilitation and Employment"
- Update Higher Level Review intake form to use "Veterans Readiness and Employment" instead of "Vocational Rehabilitation and Employment"
-Update Higher Level Review intake form to not use the question “Was an interview by the same office requested?”

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Tests pass
- [ ] Update behind feature toggle

### Testing Plan
1. Enable feature toggle `veterans_readiness`
2. Switch to intake user
3. Select Supplemental Claim as form type
4. Search for a Veteran
5. Verify benefit type option "Veterans Readiness and Employment" exists *instead of* "Vocational Rehabilitation and Employment"

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 --------|-------
Screen Shot 2021-11-03 at 9 53 37 PM|Screen Shot 2021-11-03 at 12 44 31 PM|---
Screen Shot 2021-11-05 at 10 27 25 AM|Screen Shot 2021-11-05 at 10 25 07 AM

### CODE REVIEW

File: client/constants/ISSUE_CATEGORIES.json [hunk 1]
```diff
@@ -100,6 +100,25 @@
     "Successful Closure of Case",
     "Training Facility Selection"
   ],
+  "veterans_readiness": [
```
Comment (meckila) [inline client/constants/ISSUE_CATEGORIES.json:103]: Not a fan of repetitive code, but categories for `veterans_readiness` are the same as `voc_rehab`

**Review by wdrougas-tista — APPROVED** [review]
LGTM. It looks like test is failing due to `veterans_readiness` category missing around line 1203 in `decision_reviews.yaml` file

File: client/constants/ISSUE_CATEGORIES.json (hunk 1 above)
Comment (wdrougas-tista) [reply client/constants/ISSUE_CATEGORIES.json:103]: Agreed, can hopefully be resolved down the road

File: app/controllers/api/docs/v3/decision_reviews.yaml [hunk 2]
```diff
@@ -1199,6 +1200,24 @@ components:
         - Successful Closure of Case
         - Discontinue Services
         - Interruption of Services
+        # for benefit type:  veterans_readiness
```
Comment (meckila) [inline app/controllers/api/docs/v3/decision_reviews.yaml:1203]: The same issue categories are expected twice due to update in ISSUE_CATEGORIES

File: client/app/intake/components/BenefitType.jsx [hunk 3]
```diff
@@ -2,10 +2,18 @@ import React from 'react';
...
+import { FORM_TYPES } from 'app/intake/constants';
 import { formatRadioOptions, formatSearchableDropdownOptions } from '../util';
 import SearchableDropdown from '../../components/SearchableDropdown';
 
 export default class BenefitType extends React.PureComponent {
+  updatedForms = [FORM_TYPES.SUPPLEMENTAL_CLAIM.formName, FORM_TYPES.HIGHER_LEVEL_REVIEW.formName];
+
+  benefitTypes = this.updatedForms.includes(this.props.formName) &&
+    this.props.featureToggles.updatedIntakeForms ?
+    { ...BENEFIT_TYPES, voc_rehab: 'Veterans Readiness and Employment' } :
```
Comment (nanotone) [inline client/app/intake/components/BenefitType.jsx:14]: This is a really neat way of overriding the text value!

It does raise a small question -- is it better to override the old JSON with the new text (which is what this code is doing), or to override the new JSON with the old text?

I'm trying to look ahead to when we remove this feature toggle, and wondering which way is easier, but I also think it's fine either way.

File: client/app/intake/pages/review.jsx [hunk 4]
```diff
@@ -33,10 +35,25 @@ const textAlignRightStyling = css({
...
+  };
+
+  return Object.keys(formFieldFeatureToggles).reduce((schema, toggle) => {
+    if ((featureToggles[toggle] && toggle !== 'updatedIntakeForms') ||
+    (!featureToggles[toggle] && toggle === 'updatedIntakeForms')) {
+      return schema.concat(formFieldFeatureToggles[toggle]);
+    }
+
+    return schema;
+  }, reviewHigherLevelReviewSchema);
```
Comment (nanotone) [inline client/app/intake/pages/review.jsx:51]: Oof, I imagine this change would've been a bit less bulky if we had removed the filedByVaGovHlr feature toggle first. Oh well 😅

File: spec/feature/intake/higher_level_review_spec.rb [hunk 5]
```diff
@@ -401,61 +401,66 @@
 
   let(:special_issue_reference_id) { "IAMANEPID" }
 
-  it "Creates contentions with same office special issue" do
-    Fakes::VBMSService.end_product_claim_id = special_issue_reference_id
+  context("when remove same office input toggle disabled") do
```
Comment (nanotone) [inline spec/feature/intake/higher_level_review_spec.rb:404]: You're a hero for keeping the spec for the old behavior _and_ adding a whole new one for the new behavior.

**Review by nanotone — APPROVED** [review]
Great work!

I just had one question on whether to change the constants now or later, but however you choose to answer it, this is 🚀 to merge.

File: client/app/intake/pages/review.jsx (hunk 4 above)
Comment (meckila) [reply client/app/intake/pages/review.jsx:51]: Is it safe to remove the`filedByVaGovHlr` feature toggle in another PR?

File: client/app/intake/components/BenefitType.jsx (hunk 3 above)
Comment (meckila) [reply client/app/intake/components/BenefitType.jsx:14]: I can give the second option a shot locally to see if it affects other files using BENEFIT_TYPES

File: client/app/intake/pages/review.jsx (hunk 4 above)
Comment (nanotone) [reply client/app/intake/pages/review.jsx:51]: Yep!

**Pull request merged**